src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -0,0 +1,292 @@
+        }
+
+        return new CloudantClient(props, gsonBuilder);
+    }
+
+    public ClientBuilder username(String username) {
+        this.username = username;
+        return this;
+    }
+
+    public ClientBuilder password(String password) {
+        this.password = password;
+        return this;
+    }
+
+    public ClientBuilder setGsonBuilder(GsonBuilder gsonBuilder) {
+        this.gsonBuilder = gsonBuilder;
+        return this;
+    }
+
+    /** Set the maximum number of connections to be opened in the connection pool.
+     * Defaults to {@link #MAX_CONNECTIONS}
+     **/
+    public ClientBuilder setMaxConnections(int maxConnections) {
+        this.maxConnections = maxConnections;
+        return this;
+    }
+
+    public ClientBuilder setProxyURL(URL proxyURL) {
+        this.proxyURL = proxyURL;
+        return this;
+    }
+
+    public ClientBuilder setProxyUser(String proxyUser) {
+        this.proxyUser = proxyUser;
+        return this;
+    }
+
+    public ClientBuilder setProxyPassword(String proxyPassword) {
+        this.proxyPassword = proxyPassword;
+        return this;
+    }
+
+    /**
+     * Sets whether hostname verification and certificate chain validation
+     * should be disabled.
+     *
+     * @param disabled set to true to disable or false to enable.
+     * @return the updated {@link ClientBuilder} object.
+     */
+    public ClientBuilder setSSLAuthenticationDisabled(boolean disabled) {
+        this.isSSLAuthenticationDisabled = disabled;
+        return this;
+    }
+
+    /**
+     * Specifies the SSLSocketFactory to use when connecting to CloudantDB
+     * over a <code>https</code> URL, when SSL authentication is enabled.
+     *
+     * @param factory An SSLSocketFactory, or <code>null</code> for the
+     *                default SSLSocketFactory of the JRE.
+     */
+    public ClientBuilder setAuthenticatedModeSSLSocketFactory(SSLSocketFactory factory) {
+        this.authenticatedModeSSLSocketFactory = factory;
+        return this;
+    }
+
+    public ClientBuilder interceptors(HttpConnectionInterceptor... interceptors) {
+        for(HttpConnectionInterceptor interceptor : interceptors) {
+            if (interceptor instanceof HttpConnectionRequestInterceptor) {
+                requestInterceptors.add((HttpConnectionRequestInterceptor) interceptor);
+            }
+            if (interceptor instanceof HttpConnectionResponseInterceptor) {
+                responseInterceptors.add((HttpConnectionResponseInterceptor) interceptor);
+            }
+        }
+        return this;
+    }
+
+    /** Sets the connection timeout for timeout interceptor.
+     * Defaults to {@link #CONNECTION_TIMEOUT}
+     **/
+    public ClientBuilder setConnectionTimeout(TimeoutOption connectionTimeout) {
+        this.connectionTimeout = connectionTimeout;
+        return this;
+    }
+
+    /** Sets the read timeout for timeout interceptor.
+     * Defaults to {@link #CONNECTION_TIMEOUT}
```

Comment:
The same applies as for `setConnectionTimeout()`, but this has the wrong default.